### PR TITLE
Ling changes

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

gamemode, balance and other shi

:cl:
- add: Changeling gamemode
- add: Brought changeling to secret
- tweak: Made changeling chemical regeneration static
- tweak: Made most changeling combat abilities require you to absorb people
- tweak: Made stealthy changeling abilities require no absorptions
- fix: Fixed "Escape on the evac shuttle while being person" objective
- fix: Changelings will now stay revved after transforming
- fix: Fixed armblade stinging a mouse making it drop the armblade

### CODE REVIEW

**Comment by jorgun** [comment]
need a to edit /prototypes/secret_weigth.yml if you want it to be a part of secret
and i sugest you make a "traitor + ling"  game mode

**Comment by whateverusername0** [comment]
> need a to edit /prototypes/secret_weigth.yml if you want it to be a part of secret and i sugest you make a "traitor + ling" game mode

sure

**Comment by whateverusername0** [comment]
ok great merge now ⚡ !!

**Comment by Piras314** [comment]
I get that it's going to be a playtest but those percentages are kind of ridiculous... Maybe remove the percentages and keep the gamerule so it's admin-only, it'll probably be done 50% of rounds anyway lol.

**Comment by a-person5660** [comment]
what do you think the percentages should be?

**Comment by whateverusername0** [comment]
> I get that it's going to be a playtest but those percentages are kind of ridiculous... Maybe remove the percentages and keep the gamerule so it's admin-only, it'll probably be done 50% of rounds anyway lol.

> what do you think the percentages should be?

what do you think the percentages should be?

**Comment by a-person5660** [comment]
still need to fix chem regen :trollface:

**Comment by a-person5660** [comment]
> > I get that it's going to be a playtest but those percentages are kind of ridiculous... Maybe remove the percentages and keep the gamerule so it's admin-only, it'll probably be done 50% of rounds anyway lol.
> 
> > what do you think the percentages should be?
> 
> what do you think the percentages should be?

piras said in the discord to make it zero and admins will run the mode until its stable.

**Comment by whateverusername0** [comment]
image
test failed because of some unfathomable reason idk MERGE NOW :zap:

**Comment by a-person5660** [comment]
> image test failed because of some unfathomable reason idk MERGE NOW ⚡

bro still didnt remove lings from secret, smh my head.

**Comment by whateverusername0** [comment]
> bro still didnt remove lings from secret, smh my head.

i did remove it from secret weights???

**Comment by a-person5660** [comment]
mb, read wrong commit :trollface:

**Comment by misandrie** [comment]
Shrieks need a greater cooldown and extend to all shrieks.
Fleshmend should work only when stationary.
Fleshmend shouldnt work when dead.
Regenerative stasis should demand you wait 60 seconds at least before reviving.
Amount of lings per X amount of crew should be scaled down. Role is centered around round removal and each ling starts turning 30 pop into 8 pop consistently.

**Comment by whateverusername0** [comment]
> Shrieks need a greater cooldown and extend to all shrieks.

upped it to 10 seconds

> Fleshmend should work only when stationary.

can't do that but i've upped it's cooldown to 30 seconds and made it inject some imperdrezene for slowdown (to make up for the effect of flesh mending)

> Fleshmend shouldnt work when dead.

done

> Regenerative stasis should demand you wait 60 seconds at least before reviving.

i'll just up it's entry chemical cost to 75 :trollface: 

> Amount of lings per X amount of crew should be scaled down. Role is centered around round removal and each ling starts turning 30 pop into 8 pop consistently.

sure. made it 15 people to 1 ling, like headrevs, also changed max amount of lings from 8 to 6

**Comment by whateverusername0** [comment]
ok now stasis sets your chems to 0 on entry
and exiting stasis now costs 60 chems. you have to wait a full minute to exit it now, and perhaps wait a little extra if you wanna use an ability afterwards

**Comment by misandrie** [comment]
The issue was that a ling could just get up from a dead state immediately twice at full health, this solves it

File: Resources/Prototypes/game_presets.yml [hunk 1]
```diff
@@ -161,3 +162,37 @@
   - BasicStationEventScheduler
   - GameRuleMeteorScheduler
   - BasicRoundstartVariation
+
```
Comment (Piras314) [inline Resources/Prototypes/game_presets.yml:165]: Put everything under this line into Resources/Prototypes/Goobstation/game_presets.yml

**Review by Piras314 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/game_presets.yml (hunk 1 above)
Comment (whateverusername0) [reply Resources/Prototypes/game_presets.yml:165]: done

File: Content.Shared/Goobstation/Changeling/ChangelingComponent.cs [hunk 2]
```diff
@@ -151,11 +140,8 @@ public partial struct TransformData
...
     public HumanoidAppearanceComponent Appearance;
 
-    public static bool operator ==(TransformData one, TransformData two)
-        => one.Name == two.Name && one.Fingerprint == two.Fingerprint && one.DNA == two.DNA;
-    public static bool operator !=(TransformData one, TransformData two)
-        => !(one.Name == two.Name && one.Fingerprint == two.Fingerprint && one.DNA == two.DNA);
-
-
-
+    //public static bool operator ==(TransformData one, TransformData two)
```
Comment (Piras314) [inline Content.Shared/Goobstation/Changeling/ChangelingComponent.cs:143]: It hurts my soul you aren't using multiline comments here lmao.

**Review by Piras314 — APPROVED** [review]

File: Content.Shared/Goobstation/Changeling/ChangelingComponent.cs (hunk 2 above)
Comment (whateverusername0) [reply Content.Shared/Goobstation/Changeling/ChangelingComponent.cs:143]: 😭 i removed it temporarily

File: Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl [hunk 3]
```diff
@@ -22,6 +22,7 @@ changeling-sting-fail-self = You tried to sting {CAPITALIZE(THE($target))}, but
 changeling-sting-fail-ling = Someone just tried to silently sting you!
 
 changeling-sting = You silently sting {CAPITALIZE(THE($target))}
+changeling-sting-fail-simplemob = You can't sting a simplemob with that!
```
Comment (Aidenkrz) [inline Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:25]: ???

Comment (whateverusername0) [reply Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:25]: simplemob being a mouse
how else can i name it?

Comment (misandrie) [reply Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:25]: lower being

Comment (Aidenkrz) [reply Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:25]: i know what a simplemob is, it just shouldn't say it to the player. change it to say the targets name or lesser/lower being like mary said

**Comment by CatMagic1** [comment]
Where is the pixel fix

**Comment by whateverusername0** [comment]
image
extra funny

**Comment by TurboTrackerss14** [comment]
Your cowardice in adding a blacklist to traitor role selection, meaning traitorling is not possible, has been logged under "cringe". 
(I know I am late I only just noticed the blacklist and am upset at your cowardice)

**Comment by misandrie** [comment]
Good

**Pull request merged**